ui/src/timeMachine/reducers/index.ts

```diff
@@ -811,11 +843,31 @@ export const timeMachineReducer = (
         typeof action.payload
       >
       const {fieldOptions} = action.payload
-      const {fieldOptions: prevOptions} = workingView.properties
       const properties = {
         ...workingView.properties,
-        fieldOptions: trueFieldOptions(fieldOptions, prevOptions),
+        fieldOptions,
+      }
+      const view = {...state.view, properties}
+      return {...state, view}
+    }
+
+    case 'UPDATE_FIELD_OPTION': {
```

Comment:
this seemed less heavy handed than replacing the entire array... as a side effect, it should be more resilient to data inconsistencies across the app. imagine one component updating the 3rd element of the array, while another one swaps the 3rd and 6th element.